### PR TITLE
feat: use common mocks and token calls generators

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Use createDummyAction and Token calls generator from the @aragon/osx-commons-subgraph

Task ID: [OS-999](https://aragonassociation.atlassian.net/browse/OS-999)

## Type of change

See the framework lifecylce in `packages/contracts/docs/framework-lifecycle` to decide what kind of change this pull request is.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.

[OS-999]: https://aragonassociation.atlassian.net/browse/OS-999?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [UPCOMING]
 
+### Changed
+
+- Removed token call generators and used the ones from `@aragon/sdk-commons-subgraph`
+- Used createDummyAction from `@aragon/sdk-commons-subgraph`
```
Comment (Michael-A-Heuer) [inline packages/subgraph/CHANGELOG.md:13]:
```suggestion
- Used `createDummyAction` from `@aragon/sdk-commons-subgraph`
```

File: packages/subgraph/tests/admin/admin.test.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ test('Run Admin plugin (handleProposalCreated) mappings with mock event', () =>
   );
 
   // check actions
-  for (let index = 0; index < actions.length; index++) {
+  for (let index = 0; index < [action].length; index++) {
```
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/admin/admin.test.ts:110]: Why do we loop over an array of length 1?

Comment (josemarinas) [reply packages/subgraph/tests/admin/admin.test.ts:110]: I didnt rewrite the test, I just changed from the local function to the ones in the library, adapting where necessary

Comment (Michael-A-Heuer) [reply packages/subgraph/tests/admin/admin.test.ts:110]: No worries! For me, this is an example where we can improve code quality by removing unnecessary/redundant code easily.

File: packages/subgraph/tests/helpers/method-classes.ts [hunk 3]
```diff
@@ -276,12 +273,11 @@ class ERC721ContractMethods extends ERC721Contract {
       throw new Error('Symbol is null');
     }
     // we cast to string only for stoping rust compiler complaints.
```
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/helpers/method-classes.ts:275]: Is this comment still needed? 
`createERC20TokenCalls` receives string arguments.

File: packages/subgraph/tests/helpers/method-classes.ts [hunk 4]
```diff
@@ -378,13 +378,17 @@ class ERC20ContractMethods extends ERC20Contract {
     if (!this.symbol) {
       throw new Error('Symbol is null');
     }
-    // we cast to string only for stoping rust compiler complaints.
-    createTokenCalls(
+    let supply = '10';
+    if (totalSupply) {
+      supply = totalSupply;
+    }
+    // we cast to string only for stoping rust compiler complaints.      createERC20TokenCalls(DAO_TOKEN_ADDRESS, '10', 'name', 'symbol', '6');
```
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/helpers/method-classes.ts:385]: Same here. Moreover, what is the purpose behind the `createERC20TokenCalls(DAO_TOKEN_ADDRESS, '10', 'name', 'symbol', '6');` comment?

**Review by Michael-A-Heuer — COMMENTED** [review]
Looks good. I've left some minor comments.

Thanks for removing the redundant imports along the way.
Let's also remove redundant code/comments where we see them.

File: packages/subgraph/tests/helpers/method-classes.ts (hunk 4 above)
Comment (josemarinas) [reply packages/subgraph/tests/helpers/method-classes.ts:385]: forgot to remove it 😅

File: packages/subgraph/tests/admin/admin.test.ts (hunk 2 above)
Comment (josemarinas) [reply packages/subgraph/tests/admin/admin.test.ts:110]: Yep the subgraph in particular needs a lot of love when we split it

File: packages/subgraph/tests/helpers/method-classes.ts (hunk 3 above)
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/helpers/method-classes.ts:275]:
```suggestion
    // we cast to string only for stopping rust compiler complaints.
```

File: packages/subgraph/tests/helpers/method-classes.ts [hunk 5]
```diff
@@ -378,13 +378,17 @@ class ERC20ContractMethods extends ERC20Contract {
     if (!this.symbol) {
       throw new Error('Symbol is null');
     }
+    let supply = '10';
+    if (totalSupply) {
+      supply = totalSupply;
+    }
     // we cast to string only for stoping rust compiler complaints.
```
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/helpers/method-classes.ts:385]:
```suggestion
    // we cast to string only for stopping rust compiler complaints.
```

**Review by Michael-A-Heuer — DISMISSED** [review]

**Comment by Michael-A-Heuer** [comment]
One typo left

File: packages/subgraph/tests/dao/dao_v1_0_0.test.ts [hunk 6]
```diff
@@ -260,13 +258,17 @@ describe('handleNativeTokenDeposited', () => {
...
     daoTokenContract = new ExtendedERC20Contract().withDefaultValues();
-    let totalSupply = '10';
-    daoTokenContract.mockCall_createTokenCalls(totalSupply);
+
+    createERC20TokenCalls(
+      DAO_TOKEN_ADDRESS,
+      ERC20_AMOUNT_FULL,
+      'DAO Token',
+      'DAOT',
+      '6'
```
Comment (Rekard0) [inline packages/subgraph/tests/dao/dao_v1_0_0.test.ts:267]: 1- I see you used `createERC20TokenCalls` in the extended classes,
this mean u can used `mockCall_createTokenCalls(totalSupply)` normally and you do not need to refactor here

2- why `createERC20TokenCalls`  do not return total supply as well, 
perhaps we need some refactoring in the common

File: packages/subgraph/tests/helpers/method-classes.ts [hunk 7]
```diff
@@ -341,13 +337,17 @@ class ERC20WrapperContractMethods extends ERC20WrapperContract {
...
+      supply = totalSupply;
+    }
 
-    createWrappedTokenCalls(
+    createWrappedERC20TokenCalls(
       this.id,
-      this.name as string,
-      this.symbol as string,
       this.underlyingToken,
-      totalSupply
```
Comment (Rekard0) [inline packages/subgraph/tests/helpers/method-classes.ts:350]: related to https://github.com/aragon/osx/pull/527/files#r1477929739

File: packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts [hunk 8]
```diff
@@ -53,13 +48,13 @@ import {
...
-  generateEntityIdFromAddress,
   generatePluginEntityId,
+  createDummyAction,
 } from '@aragon/osx-commons-subgraph';
-import {Address, BigInt, Bytes, log} from '@graphprotocol/graph-ts';
+import {Address, BigInt} from '@graphprotocol/graph-ts';
 import {assert, clearStore, test} from 'matchstick-as/assembly/index';
 
-let actions = createDummyActions(DAO_TOKEN_ADDRESS, '0', '0x00000000');
+let action = createDummyAction(DAO_TOKEN_ADDRESS, '0', '0x00000000');
```
Comment (Rekard0) [inline packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts:57]:
```suggestion
let actions = [createDummyAction(DAO_TOKEN_ADDRESS, '0', '0x00000000')];
```

Why don't we create the array here so we don't need to create array [action] in the rest of the test?

File: packages/subgraph/CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [UPCOMING]
 
+### Changed
+
+- Removed token call generators and used the ones from `@aragon/sdk-commons-subgraph`
```
Comment (Rekard0) [inline packages/subgraph/CHANGELOG.md:12]: Perhaps we can be more explicit here

File: packages/subgraph/tests/admin/admin.test.ts (hunk 2 above)
Comment (jordaniza) [reply packages/subgraph/tests/admin/admin.test.ts:110]: would we ever be testing multiple actions in this test?

File: packages/subgraph/tests/dao/dao_v1_3_0.test.ts [hunk 10]
```diff
@@ -250,7 +248,7 @@ describe('handleExecuted', () => {
 
   describe('ERC20 action', () => {
     beforeAll(() => {
-      createTokenCalls(DAO_TOKEN_ADDRESS, 'name', 'symbol', '6', '10');
+      createERC20TokenCalls(DAO_TOKEN_ADDRESS, '10', 'name', 'symbol', '6');
```
Comment (jordaniza) [inline packages/subgraph/tests/dao/dao_v1_3_0.test.ts:251]: Would be useful to provide some context on '10' and '6' by replacing the magic numbers with a constant at the top of the file

**Review by jordaniza — COMMENTED** [review]
Mostly echo the comments from Sarkawt. Added some minor other comments

File: packages/subgraph/tests/admin/admin.test.ts (hunk 2 above)
Comment (josemarinas) [reply packages/subgraph/tests/admin/admin.test.ts:110]: @jordaniza id does not bring too much value, I updated the test a bit tho

File: packages/subgraph/tests/dao/dao_v1_3_0.test.ts [hunk 11]
```diff
@@ -250,7 +248,15 @@ describe('handleExecuted', () => {
...
-      createTokenCalls(DAO_TOKEN_ADDRESS, 'name', 'symbol', '6', '10');
+      const totalSupply = '10';
+      const decimals = '6';
+      createERC20TokenCalls(
+        DAO_TOKEN_ADDRESS,
+        totalSupply,
+        'name',
+        'symbol',
+        decimals
+      );
```
Comment (Rekard0) [inline packages/subgraph/tests/dao/dao_v1_3_0.test.ts:259]: if the name, symbol, totalSupply & decimal are the same in this test ? why don't you make it global and reuse it everywhere?

Comment (jordaniza) [reply packages/subgraph/tests/dao/dao_v1_3_0.test.ts:259]: I agree

**Review by jordaniza — APPROVED** [review]
I would move to avoiding use of magic numbers and repeated names over time but other than that the PR looks good from what I can see.

**Review by Rekard0 — APPROVED** [review]

File: yarn.lock [hunk 12]
```diff
@@ -44,6 +48,16 @@
...
+<<<<<<< HEAD
+=======
+"@aragon/osx-commons-subgraph@^0.0.4":
+  version "0.0.4"
+  resolved "https://registry.yarnpkg.com/@aragon/osx-commons-subgraph/-/osx-commons-subgraph-0.0.4.tgz#2aa52f3089d21189c9152d2f3d14c0d7c66d129f"
+  integrity sha512-cqhusJ3HNvMx+t9lXfN+Hy/5ipefNs1Tdxe+y0GvD4qgBMVU4tCbsxOpB9U2JEJNBCzFQj4E/872FFLpIErB4w==
+  dependencies:
+    "@graphprotocol/graph-ts" "0.31.0"
+
+>>>>>>> f7ac8f54 (update commons version)
```
Comment (Michael-A-Heuer) [inline yarn.lock:60]: We have an unresolved merge conflict here

**Review by Michael-A-Heuer — CHANGES_REQUESTED** [review]
Please resolve the merge conflict in the `yarn.lock` file

File: packages/subgraph/CHANGELOG.md [hunk 13]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [UPCOMING]
 
+### Changed
+
+- Removed `createERC1155TokenCalls`, `createTokenCalls`, `createWrappedTokenCalls` and `createDummyActions` and use the equivalent functions from `@aragon/sdk-commons-subgraph`
```
Comment (Michael-A-Heuer) [inline packages/subgraph/CHANGELOG.md:12]: Make sure this is up-to-date after all the changes that have happened.

For example: `createTokenCalls` was renamed by now

File: yarn.lock (hunk 12 above)
Comment (Michael-A-Heuer) [reply yarn.lock:60]: @Rekard0 Another thing that wasn't catched by our github workflows.

Comment (josemarinas) [reply yarn.lock]: https://aragonassociation.atlassian.net/browse/OS-1066

File: packages/subgraph/tests/dao/dao_v1_0_0.test.ts [hunk 14]
```diff
@@ -993,7 +989,7 @@ describe('handleExecuted', () => {
 
   describe('ERC20 action', () => {
     beforeAll(() => {
-      createTokenCalls(DAO_TOKEN_ADDRESS, 'name', 'symbol', '6', '10');
+      createERC20TokenCalls(DAO_TOKEN_ADDRESS, '10', 'name', 'symbol', '6');
```
Comment (jordaniza) [inline packages/subgraph/tests/dao/dao_v1_0_0.test.ts:992]: Add the symbols you defined earlier

**Review by jordaniza — APPROVED** [review]
image

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**